app/client/views/activities/filters/residents/residents.js

```diff
@@ -1,54 +1,30 @@
-Template.activitiesTableResidentNamesFilter.onCreated(function () {
+Template.activitiesTableResidentNamesFilter.onCreated(function() {
   // Initialize ReactiveTable filter for residents
-  this.filter = new ReactiveTable.Filter('residentFilter', ['residentIds']);
+  this.filter = new ReactiveTable.Filter("residentFilter", ["residentIds"]);
+
+  const templateInstance = this
+  templateInstance.residentsSelectOptions = new ReactiveVar(null);
+
+  Meteor.call('residentNamesGroupedtByHomes',function(err,data){
+    templateInstance.residentsSelectOptions.set(data)
+  });
+
 });
 
 Template.activitiesTableResidentNamesFilter.events({
-   "change #resident-filter": function (event, templateInstance) {
-     // get selected resident ID
-     var residentId = $(event.target).val();
+  "change #resident-filter": function(event, templateInstance) {
+    // get selected resident ID
+    var residentId = $(event.target).val();
```

Comment:
Change this to `const` while we are working on this code. I.e. the code areas you touch should have linting improvements where possible.